stubs/urllib3/urllib3/response.pyi

```diff
@@ -1,66 +1,96 @@
 import io
-from typing import Any
+from http.client import HTTPMessage as _HttplibHTTPMessage, HTTPResponse as _HttplibHTTPResponse
+from typing import Any, Iterator, Mapping
 
-from . import _collections, exceptions
-from .connection import BaseSSLError as BaseSSLError, HTTPException as HTTPException
-from .util import response
+from urllib3.connectionpool import HTTPConnection
 
-HTTPHeaderDict = _collections.HTTPHeaderDict
-ProtocolError = exceptions.ProtocolError
-DecodeError = exceptions.DecodeError
-ReadTimeoutError = exceptions.ReadTimeoutError
-binary_type = bytes  # six.binary_type
-PY3 = True  # six.PY3
-is_fp_closed = response.is_fp_closed
+from . import HTTPConnectionPool, Retry
+from ._collections import HTTPHeaderDict
+from .connection import _TYPE_BODY
 
 class DeflateDecoder:
     def __init__(self) -> None: ...
-    def __getattr__(self, name): ...
-    def decompress(self, data): ...
+    def __getattr__(self, name: str) -> Any: ...
+    def decompress(self, data: bytes) -> bytes: ...
+
+class GzipDecoderState:
+    FIRST_MEMBER: int
+    OTHER_MEMBERS: int
+    SWALLOW_DATA: int
 
 class GzipDecoder:
     def __init__(self) -> None: ...
-    def __getattr__(self, name): ...
-    def decompress(self, data): ...
+    def __getattr__(self, name: str) -> Any: ...
+    def decompress(self, data: bytes) -> bytes: ...
+
+class BrotliDecoder:
+    def __init__(self) -> None: ...
+    def flush(self) -> bytes: ...
```

Comment:
this class is only available if `brotli` is available for import. Wasn't sure how to indicate this in stubs so just added to the allowlist for now

Reply:
Allowlist is fine. We should also add a comment in the stub file itself.